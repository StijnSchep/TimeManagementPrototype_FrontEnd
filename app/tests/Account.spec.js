import { shallowMount, createLocalVue } from "@vue/test-utils";
import Account from "../components/dashboard/AccountPage";
import Vuex from "vuex";

jest.mock('tns-core-modules/application', () => ({
  android: {
    on: jest.fn()
  },
  AndroidApplication: {
    activityBackPressedEvent: {}
  }
}));

const localVue = createLocalVue();
localVue.use(Vuex);

describe("AccountPage.vue", () => {
  let store;
  let authState;
  let authActions;
  let accountState;
  let accountActions;
  let mutations;

  beforeEach(() => {
    authState = {
      token: ""
    };
    authActions = {
      getToken: jest.fn()
    };

    accountState = {
      employee: {}
    };
    accountActions = {
      fetchEmployee: jest.fn()
    };
    mutations = {
      setEmployee: jest.fn()
    };
    store = new Vuex.Store({
      modules: {
        auth: {
          authState,
          authActions
        },
        account: {
          accountState,
          accountActions
        }
      }
    });
  });

  it("mounts and renders", async () => {
    const wrapper = shallowMount(Account, { store, localVue });
    await expect(wrapper.html()).toBeTruthy();
  });

  test("renders correctly", async () => {
    const wrapper = shallowMount(Account, { store, localVue });
    await expect(wrapper.element).toMatchSnapshot();
  });

  it("is a Vue instance", async () => {
    const wrapper = shallowMount(Account, { store, localVue });
    await expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("Expect component to exist", async () => {
    const wrapper = shallowMount(Account, { store, localVue });
    await expect(wrapper).toBeTruthy();
  });

  it("Expect component to have Page element", async () => {
    const wrapper = shallowMount(Account, { store, localVue });
    await expect(wrapper.find("Page")).toBeTruthy();
  });

  it("Expect component to have GridLayout element", async () => {
    const wrapper = shallowMount(Account, { store, localVue });
    await expect(wrapper.find("GridLayout")).toBeTruthy();
  });

  it("Expect component to have Label element", async () => {
    const wrapper = shallowMount(Account, { store, localVue });
    await expect(wrapper.find("Label")).toBeTruthy();
  });

  it("Expect component to have 23 Label elements", async () => {
    const wrapper = shallowMount(Account, { store, localVue });
    await expect(wrapper.findAll("Label").length).toBe(23);
  });

  it("data returns message", async () => {
    await expect(typeof Account.data()).toBe("object");
    await expect(Account.data().employee).toBeTruthy();
  });

  it("Expect fetchEmployee to be called", async () => {
    shallowMount(Account, { store, localVue });
    const spy = jest.spyOn(accountActions, "fetchEmployee");
    const commit = jest.fn();
    const state = jest.fn();
    const token = authActions.getToken();
    accountActions.fetchEmployee({ state, commit }, token);
    await expect(spy).toBeCalled();
  });

  it("Expect setEmployee to be called", async () => {
    shallowMount(Account, { store, localVue });
    const spy = jest.spyOn(mutations, "setEmployee");
    const state = jest.fn();
    const employee = accountState.employee;
    mutations.setEmployee({ state, employee });
    await expect(spy).toBeCalled();
  });
});
