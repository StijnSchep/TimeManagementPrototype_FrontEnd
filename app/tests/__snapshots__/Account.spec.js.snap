// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPage.vue renders correctly 1`] = `
<page-stub>
  <actionbar-stub>
    <label
      text="Account"
    />
  </actionbar-stub>
   
  <scrollview-stub>
    <stacklayout-stub
      horizontalalignment="center"
    >
      <cardview-stub
        class="picture"
      >
        <stacklayout-stub>
          <label
            class="head"
            text=" Pasfoto"
          />
           
          <label
            class="fas"
            text="fa-user-circle"
          />
        </stacklayout-stub>
      </cardview-stub>
       
      <cardview-stub
        class="card"
      >
        <stacklayout-stub
          class="card-body"
        >
          <label
            class="head"
            text=" Personalia"
          />
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Roepnaam"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Naam"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Tussenvoegsel"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Achternaam"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
        </stacklayout-stub>
      </cardview-stub>
       
      <cardview-stub
        class="card"
      >
        <stacklayout-stub
          class="card-body"
        >
          <label
            class="head"
            text=" Contactgegevens"
          />
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Postcode"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Huisnummer"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Toevoeging"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Plaats"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Telefoonnummer"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Mobiel"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" E-mail"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
        </stacklayout-stub>
      </cardview-stub>
       
      <cardview-stub
        class="card"
      >
        <stacklayout-stub
          class="card-body"
        >
          <label
            class="head"
            text=" Eigenschappen"
          />
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Geslacht"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Geboortedatum"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Geboorteplaats"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Geboorteland"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Nationaliteit"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
           
          <stacklayout-stub
            class="stack"
          >
            <label
              class="record"
              text=" Burgelijke staat"
            />
             
            <textfield-stub
              class="textfield"
              editable="false"
              text="-"
            />
          </stacklayout-stub>
        </stacklayout-stub>
      </cardview-stub>
    </stacklayout-stub>
  </scrollview-stub>
</page-stub>
`;
